### PR TITLE
Update Microsoft.ApplicationInsights.AspNetCore NuGet package to the latest stable version

### PR DESCRIPTION
**Pull request by Tianyun-G**
1.Upgrade the projects from .NET Core 2.x to .NET 6 and update Microsoft.ApplicationInsights.AspNetCore NuGet package to the latest stable version.

2.Update files:

i. src/after/DroneDelivery-after/Controllers/HomeController.cs
ii. src/after/DroneDelivery-after/DroneDelivery-after.csproj
iii. src/after/DroneDelivery-after/Program.cs
iv. src/after/DroneDelivery-after/Services/PackageServiceCaller.cs
v. src/after/DroneDelivery-after/Startup.cs
vi. src/after/DroneDelivery.Common/DroneDelivery.Common.csproj
vii. src/after/PackageService/PackageService.csproj
viii. src/before/DroneDelivery-before/Controllers/HomeController.cs
ix. src/before/DroneDelivery-before/DroneDelivery-before.csproj
x. src/before/DroneDelivery-before/Program.cs
xi. src/before/DroneDelivery-before/Startup.cs
xii. src/before/DroneDelivery.Common/DroneDelivery.Common.csproj

@scottaddie and @AlexGhiondea for notification.

### CODE REVIEW

File: src/after/DroneDelivery-after/DroneDelivery-after.csproj [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.1.1" />
+    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.20.0" />
     <PackageReference Include="Microsoft.AspNetCore.App" />
```
Comment (scottaddie) [inline src/after/DroneDelivery-after/DroneDelivery-after.csproj:9]: This metapackage should be removed when using .NET 6.0.
```suggestion
```

File: src/after/DroneDelivery-after/Services/PackageServiceCaller.cs [hunk 2]
```diff
@@ -18,7 +19,11 @@ public PackageServiceCaller(HttpClient httpClient)
 
         public async Task<PackageGen> CreatePackageAsync(PackageInfo packageInfo)
         {
-            var result = await httpClient.PutAsJsonAsync($"{packageInfo.PackageId}?code={FunctionCode}", packageInfo);
+            string jsonString = JsonConvert.SerializeObject(packageInfo);
+
+            StringContent httpContent = new StringContent(jsonString, System.Text.Encoding.UTF8, "application/json");
```
Comment (scottaddie) [inline src/after/DroneDelivery-after/Services/PackageServiceCaller.cs:24]: Prefer `var` in instantiation cases like this where the type is obvious.

```suggestion
            var httpContent = new StringContent(jsonString, System.Text.Encoding.UTF8, "application/json");
```

File: src/after/DroneDelivery-after/Controllers/HomeController.cs [hunk 3]
```diff
@@ -56,7 +57,11 @@ public IActionResult SendRequests()
 
             for (int i = 0; i < RequestCount; i++)
             {
-                tasks[i] = httpClient.PostAsJsonAsync("/api/DeliveryRequests", payload);
+                string jsonString = JsonConvert.SerializeObject(payload);
+
+                StringContent httpContent = new StringContent(jsonString, System.Text.Encoding.UTF8, "application/json");
```
Comment (scottaddie) [inline src/after/DroneDelivery-after/Controllers/HomeController.cs:62]:
```suggestion
                var httpContent = new StringContent(jsonString, System.Text.Encoding.UTF8, "application/json");
```

File: src/before/DroneDelivery-before/Controllers/HomeController.cs [hunk 4]
```diff
@@ -57,7 +58,11 @@ public IActionResult SendRequests()
 
             for (int i = 0; i < RequestCount; i++)
             {
-                tasks[i] = httpClient.PostAsJsonAsync("/api/DeliveryRequests", payload);
+                string jsonString = JsonConvert.SerializeObject(payload);
+
+                StringContent httpContent = new StringContent(jsonString, System.Text.Encoding.UTF8, "application/json");
```
Comment (scottaddie) [inline src/before/DroneDelivery-before/Controllers/HomeController.cs:63]:
```suggestion
                var httpContent = new StringContent(jsonString, System.Text.Encoding.UTF8, "application/json");
```

File: src/before/DroneDelivery-before/DroneDelivery-before.csproj [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.1.1" />
+    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.20.0" />
     <PackageReference Include="Microsoft.AspNetCore.App" />
```
Comment (scottaddie) [inline src/before/DroneDelivery-before/DroneDelivery-before.csproj:9]:
```suggestion
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

**Comment by v-xuto** [comment]
@scottaddie - We have updated, please rereview this PR. Thanks a lot.

**Review by scottaddie — APPROVED** [review]

**Comment by v-xuto** [comment]
#sign-off

**Comment by v-xuto** [comment]
#sign-off

**Comment by v-xuto** [comment]
@scottaddie  Please help merge this PR. I have added `#sign-off`, but there's no reaction here.

**Comment by v-xuto** [comment]
#sign-off

**Review by markjulmar — APPROVED** [review]

**Pull request merged**